### PR TITLE
Clean cacm collection

### PR DESCRIPTION
**Pull request by yoagauthier**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -11,24 +11,53 @@
...
+
+
+def create_common_words_set():
+    common_words = []
+
+    with open(stop_list_filename, 'r') as sl:
+        data = sl.read()
+
+    for line in data.split('\n'):
+        common_words.append(line.strip())
```
Comment (SeysT) [inline main.py]: Can be written in a more efficient way:
```py
common_words = [line.strip() for line in data.split('\n')]

File: main.py [hunk 2]
```diff
@@ -11,24 +11,53 @@
...
+
+    return common_words
```
Comment (SeysT) [inline main.py]: Can be written in a more efficient way:
```py
 return [line.strip() for line in data.split('\n')]

File: document.py [hunk 3]
```diff
@@ -1,13 +1,43 @@
...
         self.add_date = ''
         self.references = ''
         self.key_words = ''
+        self.key_words_tokenized = []
+        self.citations = ''
+
+    def tokenize_doc(self):
+        self.title_tokenized = word_tokenize(self.title)
+        self.summary_tokenized = word_tokenize(self.summary)
+        self.key_words_tokenized = word_tokenize(self.key_words)
```
Comment (SeysT) [inline document.py]:
```py
for attr in Document.tokenized_fields:
    setattr(self, attr, word_tokenize(getattr(self, attr.replace('_tokenized', '')))

File: main.py [hunk 4]
```diff
@@ -11,24 +11,53 @@
...
+        if line.startswith('.I'):
+            if current_id:
+                document_collection[current_id].tokenize_doc()
+                document_collection[current_id].clean_tokens(common_words)
+            current_id = line.replace('.I ', '')
+            document_collection[current_id] = Document(current_id)
+        elif line in fields_mapping.keys():
+            attr = fields_mapping[line]
+        else:
+            attr_value = getattr(document_collection[current_id], attr)
```
Comment (SeysT) [inline main.py]:
```py
attr_value = getattr(document_collection[current_id], attr, None)

File: document.py [hunk 5]
```diff
@@ -1,13 +1,43 @@
...
+            if token in common_words or token in Document.signs_to_remove:
+                self.title_tokenized.remove(token)
+
+        for token in self.summary_tokenized:
+            if token in common_words or token in Document.signs_to_remove:
+                self.summary_tokenized.remove(token)
+
+        for token in self.key_words_tokenized:
+            if token in common_words or token in Document.signs_to_remove:
+                self.key_words_tokenized.remove(token)
```
Comment (SeysT) [inline document.py]:
```py
for attr in Document.tokenized_fields:
    new_attr_value = [
        token if not (token in common_words or token in Document.signs_to_remove) for token in getattr(self, attr)
    ]
    setattr(self, attr, new_attr_value)

File: document.py [hunk 6]
```diff
@@ -1,13 +1,43 @@
...
+        self.title_tokenized = []
         self.summary = ''
+        self.summary_tokenized = []
         self.publication_date = ''
         self.authors = ''
         self.add_date = ''
         self.references = ''
         self.key_words = ''
+        self.key_words_tokenized = []
+        self.citations = ''
```
Comment (SeysT) [inline document.py:32]: I think we only need the first line of the constructor (if all other modification apply):
```py
def __init__(self, id):
    self.id = id

File: main.py [hunk 7]
```diff
@@ -11,24 +11,53 @@
     '.N': 'add_date',
     '.X': 'references',
     '.K': 'key_words',
+    '.C': 'citations',
 }
```
Comment (SeysT) [inline main.py]: Maybe move this dict in Document ?

File: main.py (hunk 2 above)
Comment (yoagauthier) [reply main.py]: Agreed

File: main.py (hunk 4 above)
Comment (yoagauthier) [reply main.py]: Agreed

File: main.py (hunk 7 above)
Comment (yoagauthier) [reply main.py]: Agreed

File: document.py (hunk 5 above)
Comment (yoagauthier) [reply document.py]: I find the one-liner quite hard to read with loop & condition. What about : 
```python
for attr in Document.tokenized_fields:
    for token in getattr(self, attr):
        if token in common_words or token in Document.signs_to_remove:
            getattr(self, attr).remove(token)
```

File: document.py (hunk 3 above)
Comment (yoagauthier) [reply document.py]: Agreed

**Pull request merged**